src/providers/digitalocean/mod.rs

```diff
@@ -121,7 +121,7 @@ fn parse_attrs(data: &Metadata) -> Result<Vec<(String,String)>> {
         for a in ifaces {
```

Comment:
Note: this should `enumerate()` to also track the loop counter. I didn't try but a straight fix should look like:
```
for (i, a) in ifaces.iter().cloned().enumerate() { ... }
```

(there is also probably a bit too much cloning going on)